src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -148,33 +149,53 @@ void reconnect() throws TimeoutException, ExecutionException, InterruptedExcepti
         }
     }
 
-    @SuppressWarnings("PMD.NullAssignment")
+
     protected synchronized CompletableFuture<Boolean> connect() {
         if (connection != null) {
             return CompletableFuture.completedFuture(true);
         }
+        // For the initial connect, client connects with cleanSession=true and disconnects.
+        // This deletes any previous session information maintained by IoT Core.
+        // For subsequent connects, the client connects with cleanSession=false
+        if (initalConnect.get()) {
+            try {
+                establishConnection(true).get(getTimeout(), TimeUnit.MILLISECONDS);
+                disconnect();
```

Comment:
disconnect can sometimes take a very long time, or even timeout. You may need to do some manual testing or talk to the SDK team to figure that one out.

Reply:
Manually tested by performing connect -> publish/subscribe -> disconnect actions in a loop. Did not see disconnect timing out. Connect operation did fail a couple of times (twice in 100 attempts) but increasing the timeout did not fix that issue. Even when the connect failed, when it was retried for the next publish operation it went through.